File: homework02/Random_timerCsharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Random_timerCsharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        public Random r = new Random();

        private void button2_Click(object sender, EventArgs e)
        {
            this.timer1.Stop();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.timer1.Start();
        }

        public int i;
        public double CurrentArithmeticMean = 0;

        private void timer1_Tick(object sender, EventArgs e)
        {
            i += 1;
            String nameOfExam = "Exam" + i;
            int gradeOfExam = r.Next(18, 31);
            CurrentArithmeticMean = CurrentArithmeticMean + (gradeOfExam - CurrentArithmeticMean) / i;
            this.richTextBox1.AppendText(nameOfExam.PadRight(10) + gradeOfExam + " Current Mean: " + CurrentArithmeticMean + "\n");

        }
    }
}

File: homework02/Random_Timer/Form1.vb
Public Class Form1

    Public R As New Random
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.Timer1.Start()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Timer1.Stop()
    End Sub

    Public CurrentArithmeticMean As Double = 0

    Public i As Integer
    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        i = i + 1
        Dim nameOfExam = "Exam" & i
        Dim GradeOfExam = Me.R.Next(18, 31)
        ' Me.RichTextBox1.AppendText(nameOfExam.PadRight(10) & GradeOfExam & Environment.NewLine)
        CurrentArithmeticMean = CurrentArithmeticMean + (GradeOfExam - CurrentArithmeticMean) / i
        Me.RichTextBox1.AppendText(nameOfExam.PadRight(10) & GradeOfExam & "    " & "Current Mean: " & CurrentArithmeticMean & Environment.NewLine)
    End Sub
End Class
